QueueDown/PipesAndChannels5.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System.Buffers;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.IO.Pipelines;
+using System.Threading.Channels;
+
+partial class Program
+{
+    // for debugging
+    static Output[] s_outputs = new Output[50];
+
+    public static PipeReader Pipes5(Counter<long> counter, List<Task> tasks)
+    {
+        //var pool = new PinnedBlockMemoryPool();
+        var pool = MemoryPool<byte>.Shared;
+        var channel = Channel.CreateBounded<Output>(new BoundedChannelOptions(50) 
+        {
+            SingleReader = true
+        });
+
+        // Multiple producers
+        for (int i = 0; i < 50; i++)
+        {
+            var producerPipe = new Pipe(new(pool, minimumSegmentSize:64*1024 ));
```

Comment:
How relevant is this to the performance improvement?

Reply:
a lot : )
```
buffer size      pauseWriterThreshold        transfer-rate KiB/s
256KB             256KB                              78M
64KB              default (64K?)                     65M
4KB               default (64K?)                     17M
```

Reply:
Now I have to play with tuning my implementation's buffers 😀